projects/ngx-feature-flag/src/lib/service/feature-toggles.service.spec.ts

```diff
@@ -175,34 +169,20 @@ describe('FeatureToggles service:', () => {
           id: 'Deployments.featureB'
         }
       ]};
-    mockService.connections.subscribe((connection: any) => {
-      connection.mockRespond(new Response(
-        new ResponseOptions({
-          body: JSON.stringify(expectedResponse),
-          status: 200
-        })
-      ));
-    });
     // when
     togglesService.getFeaturesPerPage('Deployments').subscribe(features => {
       // then
-      expect(features.length).toEqual(2);
+      expect(features as Feature[]).toBeTruthy();
+      expect((features as Feature[]).length).toEqual(2);
       expect((features[0] as Feature).id).toEqual(expectedResponse.data[0].id);
       expect((features[0] as Feature).attributes['name']).
       toEqual(expectedResponse.data[0].attributes['name']);
     });
+    const req = httpTestingController.expectOne(request => request.method === 'GET' && request.url === `${url}/features`);
```

Comment:
As a general comment on all these tests using the new HttpClient and testing controller - should there not be any "controller.verify()" calls in here to assert that no additional requests are being fired other than the ones explicitly expected and tested for?

Reply:
Also, do we know for certain that the test controller, HttpClient, and RX Scheduler in use are all synchronous? If not then there's the possibility that the subscribers in these tests don't actually get invoked within the lifecycle of the individual test runs, which can lead to false positives as well as apparent failures in the wrong tests when failures do occur. We've chased these down before and it's no fun...

Using `.first()` to handle subscription cleanup and `done: DoneFn` to ensure that the test does not exit before the subscriber is invoked seems to work well in fabric8-ui.

Reply:
@andrewazores there is:
```
  afterEach(() => {
    httpTestingController.verify();
  });
```

Reply:
Ah right, didn't see that, sorry. I still have my other concerns regarding `DoneFn` etc. though.